### PR TITLE
Fix/missing dep safe tests

### PR DESCRIPTION
**Pull request by hredestig**
make unit-tests run also without dependencies cplex and redis. fix a couple of other minor issues

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,12 +23,7 @@ addons:
 before_install:
 - pip install pip --upgrade
 - pip install codecov
-- curl -L $SECRET_CPLEX_LINK -o cplex.tar.gz
-- tar xvf cplex.tar.gz
-- if [[ $TRAVIS_PYTHON_VERSION == "3.4" ]]; then cd "cplex/python/3.4/x86-64_linux/"; fi
-- if [[ $TRAVIS_PYTHON_VERSION == "2.7" ]]; then cd "cplex/python/2.6/x86-64_linux/"; fi
-- pip install .
-- cd $TRAVIS_BUILD_DIR
+- if [[ $TRAVIS_PULL_REQUEST != "false" ]]; then bash ./.travis/install_cplex.sh; fi
```
Comment (sgalkina) [inline .travis.yml]: can it be `== "true"`?

File: cameo/strain_design/heuristic/evolutionary/multiprocess/migrators.py [hunk 2]
```diff
@@ -15,7 +15,11 @@
 from __future__ import absolute_import, print_function
 
 import six.moves.queue
-from cameo.parallel import RedisQueue
+try:
+    from cameo.parallel import RedisQueue
+except ImportError:
+    RedisQueue = None
+    pass
```
Comment (sgalkina) [inline cameo/strain_design/heuristic/evolutionary/multiprocess/migrators.py]: `pass` is not needed

File: cameo/util.py [hunk 3]
```diff
@@ -436,7 +436,7 @@ def generate_colors(n):
     color_map = {}
     for i in range(n):
         rgb = colorsys.hsv_to_rgb(*hsv_tuples[i])
-        color = tuple([rgb[0] * 256, rgb[1] * 256, rgb[2] * 256])
+        color = tuple([int(rgb[0] * 256), int(rgb[1] * 256), int(rgb[2] * 256)])
```
Comment (sgalkina) [inline cameo/util.py]:
```python
tuple(int(i*256) for i in rgb)
```
minor, just a bit shorter :)

File: cameo/strain_design/heuristic/evolutionary/multiprocess/migrators.py (hunk 2 above)
Comment (hredestig) [inline cameo/strain_design/heuristic/evolutionary/multiprocess/migrators.py]: cheers, silly typo

**Comment by hredestig** [comment]
codecov with double testing strategies (with/without cplex) makes it tricky to enforce coverage diff >= 0. Alternatives I see
- make all unit-tests that run cameo code use glpk combined with annotation in coveragerc to not count cplex specific code
- make a special default branch to form pull requests to, which was coverage measured without cplex (i.e. devel) diff is then reported to that branch as base

any other ideas?

**Comment by hredestig** [comment]
Alright, so favored solution is that travis will *never* run non-free (cplex) tests for devel and that all contributions should be based on that branch to compare against that lower test coverage. Maintainers can merge to new branch `devel-nofree` to have all tests run and complete coverage measured

**Pull request merged**